server/channels/store/sqlstore/file_info_store.go

```diff
@@ -726,10 +727,13 @@ func (fs SqlFileInfoStore) GetFilesBatchForIndexing(startTime int64, startFileID
 			},
 		}).
 		OrderBy("FileInfo.CreateAt ASC, FileInfo.Id ASC").
-		Limit(uint64(limit)).
-		ToSql()
+		Limit(uint64(limit))
+
+	if !includeDeleted {
+		query = query.Where(sq.Eq{"FileInfo.DeleteAt": 0})
+	}
```

Comment:
Should we actually include deleted rows? From my testing, while we soft delete the `FileInfo` records, we don't actually delete the underlying files, so they might still need to be migrated in order to be accessible (if the soft deletion were reversed). This would also reduce the set of files we need to touch 🎉